source/adapters/cuda/enqueue.cpp

```diff
@@ -484,6 +484,133 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueCooperativeKernelLaunchExp(
                                numEventsInWaitList, phEventWaitList, phEvent);
 }
 
+UR_APIEXPORT ur_result_t UR_APICALL urEnqueueKernelLaunchCustomExp(
+    ur_queue_handle_t hQueue, ur_kernel_handle_t hKernel, uint32_t workDim,
+    const size_t *pGlobalWorkSize, const size_t *pLocalWorkSize,
+    uint32_t numAttrsInLaunchAttrList,
```

Comment:
Thanks, yeah I think I should change this to make it consistent with the other properties.

Reply:
After some test implementations  and discussion we decided that whilst it would be nice to have a consistent properties protocol, the existing way is the best option atm:

- the most common way of passing properties via a linked list of descriptor and properties would work, but there are a couple of small things that makes it not exactly natural for this interface/property type combination. There are existing interfaces in ur that also pass properties in different ways
- The existing way is the most natural mapping to how cuda uses these properties (attributes)
- This is experimental and we could potentially update it later: Once we know about other backend requirements we will probably have to update some aspects of it anyway.

However I have changed the naming from attributes -> properties, because in UR attributes is used for things like device queries, and "properties" is consistent with UR naming for passing in properties to functions.